### PR TITLE
Enable proxy for optOutJs

### PR DESCRIPTION
**Pull request by Skywalker-11**
Enables the use of the external js file for the opt-out page https://github.com/matomo-org/matomo/pull/12873

### CODE REVIEW

**Comment by mattab** [comment]
Thanks for the PR. We will close it for now (see comment in https://github.com/matomo-org/matomo/pull/12873).

File: matomo-proxy.php [hunk 1]
```diff
@@ -5,7 +5,10 @@
 $path = 'index.php';
 
 $SUPPORTED_METHODS = [
-    'CoreAdminHome.optOut',
+    'CoreAdminHome.optOut'
+];
+$VALID_FILES = [
+    'plugins/CoreAdminHome/javascripts/optOut.js'
 ];
```
Comment (diosmosis) [inline matomo-proxy.php]: 👍 for creating a list of `$VALID_FILES`

File: proxy.php [hunk 2]
```diff
@@ -56,8 +56,8 @@
 // the HTTP response headers captured via fopen or curl
 $httpResponseHeaders = [];
 
-// 1) PIWIK.JS PROXY: No _GET parameter, we serve the JS file
-if (empty($_GET) && empty($_POST)) {
+// 1) PIWIK.JS PROXY: No _GET parameter, we serve the JS file; or we serve a requested js file
+if ((empty($_GET) && empty($_POST)) || (isset($filerequest) && substr($filerequest, -3) !== false && substr($filerequest, -3) === '.js')) {
```
Comment (diosmosis) [inline proxy.php]: This can be simplified to `isset($filerequest) && substr($filerequest, -3) === '.js'`

File: tests/ProxyTest.php [hunk 3]
```diff
@@ -254,6 +254,15 @@ public function test_indexphp_requests_are_proxied_correctly()
         $this->assertEquals($expected, $responseBody);
     }
```
Comment (diosmosis) [inline tests/ProxyTest.php]: Could you modify the `test_indexphp_requests_are_proxied_correctly` test so it also tests the URL rewriting code you added?

**Pull request merged**